mlir/lib/Dialect/LLVMIR/IR/LLVMLinkerInterface.cpp

```diff
@@ -505,8 +505,11 @@ Operation *LLVM::LLVMSymbolLinkerInterface::appendGlobals(llvm::StringRef glob,
     return appendGlobalStructors<LLVM::GlobalDtorsOp>(state);
 
   const auto &globs = append.lookup(glob);
-  if (globs.empty())
+  if (globs.empty()) {
+    if (auto found = summary.find(glob); found != summary.end())
+      return state.clone(found->second);
     return nullptr;
+  }
```

Comment:
Do you have a test-case for this? Unless it's too complicated.
Feels like we are chasing bugs in this part, so it would be good to have regression tests